services/QuillLMS/app/serializers/admin/teacher_serializer.rb

```diff
@@ -4,24 +4,23 @@
 class Admin::TeacherSerializer < ApplicationSerializer
   include Rails.application.routes.url_helpers
 
-  attributes :id, :name, :email, :school,
-            :links,
+  ADMIN = 'Admin'
+  TEACHER = 'Teacher'
+
+  attributes :id, :name, :email, :last_sign_in, :schools,
             :number_of_students,
             :number_of_activities_completed,
             :time_spent,
             :has_valid_subscription
 
   type :teacher
 
-  def school
-    object.try(:school).try(:name)
-  end
-
-  def links
-    [
-      { name: "Teacher Access", path: admin_sign_in_classroom_manager_user_path(object) },
-      { name: "Premium Reports", path: admin_sign_in_progress_reports_user_path(object) },
-    ]
+  def schools
+    [object&.school].concat(object&.administered_schools).compact.uniq.map do |school|
+      school_hash = { name: school.name, id: school.id }
+      school_hash[:role] = SchoolsAdmins.find_by(school: school, user: object) ? ADMIN : TEACHER
```

Comment:
I would use `SchoolsAdmins.exists?(school: school, user: object)` since it is more performant and also communicates that you don't need the object later.